File: backend/app/core/supabase.py
from supabase import create_client
from dotenv import load_dotenv
import os

load_dotenv()

SUPABASE_URL = "https://uzgpjmvmoncwmfkbeknr.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InV6Z3BqbXZtb25jd21ma2Jla25yIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MzYzMjA1NTAsImV4cCI6MjA1MTg5NjU1MH0.UXEuxEAzbRzS3sg9F_LWhrVZNuAEpPbU4L_FgxQ90Oc"

supabase = create_client(SUPABASE_URL, SUPABASE_KEY)

# Helper functions for common operations
async def get_user_by_email(email: str):
    response = supabase.table('users').select('*').eq('email', email).execute()
    return response.data[0] if response.data else None

async def get_user_by_username(username: str):
    response = supabase.table('users').select('*').eq('username', username).execute()
    return response.data[0] if response.data else None

async def create_user(user_data: dict):
    response = supabase.table('users').insert(user_data).execute()
    return response.data[0] if response.data else None

async def get_detections_by_user(user_id: str, limit: int = 100):
    response = supabase.table('detections').select('*').eq('user_id', user_id).limit(limit).execute()
    return response.data

async def create_detection(detection_data: dict):
    response = supabase.table('detections').insert(detection_data).execute()
    return response.data[0] if response.data else None 

File: frontend/src/services/supabase.ts
import { createClient } from '@supabase/supabase-js';

const supabaseUrl = 'https://uzgpjmvmoncwmfkbeknr.supabase.co';
const supabaseKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InV6Z3BqbXZtb25jd21ma2Jla25yIiwicm9sZSI6ImFub24iLCJpYXQiOjE3MzYzMjA1NTAsImV4cCI6MjA1MTg5NjU1MH0.UXEuxEAzbRzS3sg9F_LWhrVZNuAEpPbU4L_FgxQ90Oc';

export const supabase = createClient(supabaseUrl, supabaseKey);

interface UserData {
  username: string;
  full_name: string;
  department: string;
  role: string;
}

interface Detection {
  id: string;
  user_id: string;
  detection_type: 'image' | 'text' | 'multimodal';
  content_path: string;
  threat_score: number;
  confidence_score: number;
  analysis_results: {
    details: string;
    indicators?: string[];
    severity?: string;
  };
  threat_category: string;
  remediation_suggestions: {
    actions: string[];
    priority?: string;
    description?: string;
  };
  created_at: string;
  updated_at: string;
}

// Auth helpers
export const auth = {
  signUp: async (email: string, password: string, userData: UserData) => {
    const { data, error } = await supabase.auth.signUp({
      email,
      password,
      options: {
        data: userData,
      },
    });
    if (error) throw error;
    return data;
  },

  signIn: async (email: string, password: string) => {
    const { data, error } = await supabase.auth.signInWithPassword({
      email,
      password,
    });
    if (error) throw error;
    return data;
  },

  signOut: async () => {
    const { error } = await supabase.auth.signOut();
    if (error) throw error;
  },

  getUser: async () => {
    const { data: { user }, error } = await supabase.auth.getUser();
    if (error) throw error;
    return user;
  },
};

// Detection helpers
export const detection = {
  createDetection: async (detectionData: Omit<Detection, 'id' | 'created_at' | 'updated_at'>) => {
    const { data, error } = await supabase
      .from('detections')
      .insert(detectionData)
      .select()
      .single();
    if (error) throw error;
    return data as Detection;
  },

  getHistory: async () => {
    const { data, error } = await supabase
      .from('detections')
      .select('*')
      .order('created_at', { ascending: false });
    if (error) throw error;
    return data as Detection[];
  },

  getStats: async () => {
    const { data, error } = await supabase
      .from('detections')
      .select('*');
    if (error) throw error;
    return data as Detection[];
  },
}; 